extensions/iframe/common/client.mjs

```diff
@@ -0,0 +1,73 @@
+// Copyright 2021 HITCON Online Contributors
+// SPDX-License-Identifier: BSD-2-Clause
+
+import Overlay from '/static/sites/game-client/ui/overlay.mjs';
+import Modal from '/static/sites/game-client/ui/modal.mjs';
+import OverlayPosition from '/static/sites/game-client/ui/overlay-position.mjs';
+import {MapCoord} from '/static/common/maplib/map.mjs';
+
+class IframeOverlay extends Overlay {
+  constructor(mainUI) {
+    const dom = document.getElementById('iframe-overlay');
+    super(mainUI, dom);
+    this.iframe = dom.querySelector('iframe');
+    this.hide();
+    this.isOpen = false;
+  }
+
+  display(src, attributes = {}) {
+    if (this.isOpen) return;
+    this.iframe.src = src;
+    this.show(OverlayPosition.LEFT_BOTTOM);
+    this.isOpen = true;
+  }
+
+  close() {
+    this.hide();
+    this.isOpen = false;
+  }
+}
+
+class IframeModal extends Modal {
```

Comment:
Not sure if need IframeModal. If we indeed use IframeModal, then the player can't move when the IframeModal shows up. Currently we don't have a use case for this, but it doesn't hurt to leave it there with a comment on it.

Reply:
I think there will be a close button when IframModal is active. I decide to remove it  and wait for the real cases.